src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -142,6 +142,39 @@ public void Commit()
         #region Helpers
 
         private void AddRuleset()
+        {
+            TrySetNonConditionalRuleSet();
+            SetConditionalRuleSets();
+        }
+
+        /// <summary>
+        /// Try to set a ruleset without specifying a configuration (e.g. Debug / Release).
+        /// No changes are made if the project already has a ruleset.
+        /// </summary>
+        private void TrySetNonConditionalRuleSet()
+        {
+            var hasUserSpecifiedRuleSet = propertyInformationMap.Keys.Any(x => !IsDefaultMicrosoftRuleSet(x.Value as string));
+
+            if (hasUserSpecifiedRuleSet)
+            {
+                // We could've done this for all projects that have configurations that point to the same ruleset but we don't want to create noise for existing users.
+                return;
+            }
+
+            var ruleSetFullFilePath = propertyInformationMap.Values.FirstOrDefault()?.NewRuleSetFilePath;
+
+            if (!string.IsNullOrEmpty(ruleSetFullFilePath))
```

Comment:
This IF will never happen in reality, as VS will return the default value even if the project does not specify it. 
But it caused some tests to fail when they didn't set it, and it caused other tests to fail when I did set it... So it was easier to just add the IF :(

Reply:
Strictly speaking the default value is being returned because of the contents of the standard MS MSBuild targets, not VS. So far those targets always provide a default.

Reply:
...although as commented above, we don't need to be accessing the property information map to get the ruleset path anyway.